astropy/table/column.py

```diff
@@ -935,9 +935,17 @@ def _make_compare(oper):
         """
 
         def _compare(self, other):
+            # Special case to work around #6838.  Other combinations work OK,
+            # see tests.test_column.test_unicode_sandwich_compare().  In this
+            # case just swap self and other.
+            if (isinstance(self, MaskedColumn) and self.dtype.kind == 'U' and
+                    isinstance(other, MaskedColumn) and other.dtype.kind == 'S'):
+                self, other = other, self
```

Comment:
This is OK only for `==` and `!=`, not for the other comparisons...

Reply:
Good point.  I actually had written a `<` compare into the tests but then deleted.

Reply:
Actually, have you tried a simpler option, of just returning `NotImplemented`? python should then automatically call the reverse operator:
```
In [6]: class A:
   ...:     def __lt__(self, other):
   ...:         return NotImplemented
   ...:     def __gt__(self, other):
   ...:         return 'Yes'
   ...:     

In [7]: a = A()

In [8]: a > a
Out[8]: 'Yes'

In [9]: a < a
Out[9]: 'Yes'
```

I think this would not work for `__eq__` and `__ne__`, though, since those do not have a different swapped version. So perhaps the clarity of this implementation still is preferable.

Reply:
I implemented the first suggestion which does look cleaner (and saves some memory and 98 ns per compare).  On the second idea I think that code simplicity wins.